crates/networking/rpc/clients/eth/mod.rs

```diff
@@ -667,48 +605,59 @@ impl EthClient {
         }
     }
 
+    pub async fn set_gas_for_wrapped_tx(
+        &self,
+        wrapped_tx: &mut WrappedTransaction,
+        from: Address,
+    ) -> Result<(), EthClientError> {
+        let mut transaction = match wrapped_tx {
+            WrappedTransaction::EIP4844(wrapped_eip4844_transaction) => {
+                GenericTransaction::from(wrapped_eip4844_transaction.clone().tx)
+            }
+            WrappedTransaction::EIP1559(eip1559_transaction) => {
+                GenericTransaction::from(eip1559_transaction.clone())
+            }
+            WrappedTransaction::L2(privileged_l2_transaction) => {
+                GenericTransaction::from(privileged_l2_transaction.clone())
+            }
+        };
+
+        transaction.from = from;
+        let gas_limit = self.estimate_gas(transaction).await?;
```

Comment:
I think this could be reduced to calling `estimate_gas_for_wrapped_tx`.

```suggestion
        let gas_limit = self.estimate_gas_for_wrapped_tx(wrapped_tx, from).await?;
```